rhizome/host/lib/cert_server_setup.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require_relative "../../common/lib/util"
+require_relative "../../common/lib/arch"
+require_relative "vm_path"
+require "fileutils"
+require "json"
+
+class CertServerSetup
+  def initialize(vm_name)
+    @vm_name = vm_name
+  end
+
+  def vp
+    @vp ||= VmPath.new(@vm_name)
+  end
+
+  def cert_folder
+    vp.q_cert
+  end
+
+  def cert_path
+    "#{cert_folder}/cert.pem"
+  end
+
+  def key_path
+    "#{cert_folder}/key.pem"
+  end
+
+  def service_name
+    "#{@vm_name}-metadata-endpoint"
+  end
+
+  def service_file_path
+    "/etc/systemd/system/#{service_name}.service"
+  end
+
+  def server_main_path
+    File.join("", "opt", "metadata-endpoint")
+  end
+
+  def vm_server_path
+    File.join(cert_folder, "metadata-endpoint")
+  end
+
+  def package_url
+    if Arch.arm64?
+      "https://github.com/ubicloud/metadata-endpoint/releases/download/v0.1.3/metadata-endpoint_Linux_arm64.tar.gz"
+    elsif Arch.x64?
+      "https://github.com/ubicloud/metadata-endpoint/releases/download/v0.1.3/metadata-endpoint_Linux_x86_64.tar.gz"
+    else
+      fail "BUG: unexpected architecture"
+    end
```

Comment:
nit: `Arch` has `.render` helper. It makes reduce the number of branches to tests. 
https://github.com/ubicloud/ubicloud/blob/a8952c1f15cba9a15f33d69b3f727494cab64a4b/rhizome/common/lib/arch_class.rb#L25

```suggestion
    Arch.render(
        x64: "https://github.com/ubicloud/metadata-endpoint/releases/download/v0.1.3/metadata-endpoint_Linux_x86_64.tar.gz",
        arm64: "https://github.com/ubicloud/metadata-endpoint/releases/download/v0.1.3/metadata-endpoint_Linux_arm64.tar.gz")
```